test/int/featureBasedSuits/hitCountBreakpointsDifferentOperators.test.ts

```diff
@@ -0,0 +1,124 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+/*
+ * Hit count breakpoints' scenarios
+ * Hit count breakpoint syntax: (>|>=|=|<|<=|%)?\s*([0-9]+)
+ */
+
+import * as _ from 'lodash';
+import { puppeteerSuite, puppeteerTest } from '../puppeteer/puppeteerSuite';
+import { reactWithLoopTestSpecification } from '../resources/resourceProjects';
+import { BreakpointsWizard as BreakpointsWizard } from '../wizards/breakpoints/breakpointsWizard';
+import { expect } from 'chai';
+
+puppeteerSuite('Hit count breakpoints combinations', reactWithLoopTestSpecification, (suiteContext) => {
+    interface IConditionConfiguration {
+        condition: string; // The condition for the hit count breakpoint
+        iterationsExpectedToPause: number[]; // In which iteration numbers it should pause (e.g.: 1st, 5th, 12th, etc...)
+    }
+
+    // * Hit count breakpoint syntax: (>|>=|=|<|<=|%)?\s*([0-9]+)
+    const manyConditionsConfigurations: IConditionConfiguration[] = [
+        { condition: '=     0', iterationsExpectedToPause: [] },
+        { condition: '= 1', iterationsExpectedToPause: [1] },
+        { condition: '= 2', iterationsExpectedToPause: [2] },
+        { condition: '= 12', iterationsExpectedToPause: [12] },
+        { condition: '> 0', iterationsExpectedToPause: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10] },
+        { condition: '> 1', iterationsExpectedToPause: [2, 3, 4, 5, 6, 7, 8, 9, 10] },
+        { condition: '>\t2', iterationsExpectedToPause: [3, 4, 5, 6, 7, 8, 9, 10] },
+        { condition: '> 187', iterationsExpectedToPause: [188, 189, 190, 191] },
+        { condition: '>=   0', iterationsExpectedToPause: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10] },
+        { condition: '>= 1', iterationsExpectedToPause: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10] },
+        { condition: '>= 2', iterationsExpectedToPause: [2, 3, 4, 5, 6, 7, 8, 9, 10] },
+        { condition: '>= 37', iterationsExpectedToPause: [37, 38, 39] },
```

Comment:
How many iterations are there? I don't follow why this only goes up to 39 when others are testing up to higher numbers

Reply:
If I added a 40 after the 39, the test would test one more iteration and work properly.

There are 200 iterations in the loop. I just test enough of them to make sure that the condition is working as expected.

I coded the test like this on purpose, because hitting 200 breakpoints for the >= 0 condition is probably not worth it...

Reply:
But looking at that, and `{ condition: '<= 1', iterationsExpectedToPause: [1] },`, I don't see how it's testing that you are _not_ pausing on `2`, but the first test isn't testing that you are not pausing on `40`, if that makes sense.

Reply:
If it pauses on `2`, will the test fail? It doesn't look like it

Reply:
That's a great point. Thank you for catching it :)

Reply:
Added a new noMorePausesAfterwards parameter to verify that we don't pause on the 2nd iteration for <= 1 and other similar cases.